examples/extensions/remove_entire_sitelink_campaign_extension_setting.py

```diff
@@ -0,0 +1,230 @@
+    return mutate_operation
+
+
+def _get_all_sitelink_extension_feed_items(
+    client, ga_service, customer_id, campaign_id
+):
+    """Gets all sitelink extension feed items associated to the specified
+    campaign extension setting.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        ga_service: the Google Ads API service client.
+        customer_id: the client customer ID.
+        campaign_id: the ID of the campaign to get the sitelink extension feed
+            items from.
+
+    Returns:
+        An array of str resource names of extension feed items.
+    """
+    campaign_resource_name = client.get_service(
+        "CampaignService", version="v6"
+    ).campaign_path(customer_id, campaign_id)
+    extension_type_enum = client.get_type("ExtensionTypeEnum", version="v6")
+    extension_type_name = extension_type_enum.ExtensionType.Name(
+        extension_type_enum.ExtensionType.SITELINK
+    )
+
+    # Construct the query.
+    query = f"""
+        SELECT
+            campaign_extension_setting.campaign,
+            campaign_extension_setting.extension_type,
+            campaign_extension_setting.extension_feed_items
+        FROM campaign_extension_setting
+        WHERE campaign_extension_setting.campaign = '{campaign_resource_name}'
+        AND campaign_extension_setting.extension_type =
+            '{extension_type_name}'"""
+
+    # Issue a search request using streaming.
+    response = ga_service.search_stream(customer_id, query=query)
+    extension_feed_item_resource_names = []
+    # Iterate through each row and append the extension feed item resource
+    # names to the return array.
+    for batch in response:
+        for row in batch.results:
+            extension_feed_item_resource_names += (
+                row.campaign_extension_setting.extension_feed_items
+            )
+
+    if len(extension_feed_item_resource_names) == 0:
+        raise Exception(
+            "The specified campaign does not contain a sitelink "
+            "campaign extension setting."
+        )
+
+    return extension_feed_item_resource_names
+
+
+def _create_extension_feed_item_mutate_operations(
+    client, extension_feed_item_resource_names
+):
+    """Creates MutateOperations for the sitelink extension feed items that will
+    be removed.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        extension_feed_item_resource_names: the extension feed item resource
+            names.
+
+    Returns:
+        An array of MutateOperations for the extension feed items.
+    """
+    mutate_operations = []
+    # Create a MutateOperation for each extension feed item to remove.
+    for resource_name in extension_feed_item_resource_names:
+        mutate_operation = client.get_type("MutateOperation", version="v6")
+        mutate_operation.extension_feed_item_operation.remove = resource_name
+        mutate_operations.append(mutate_operation)
```

Comment:
why are you mixing `<list>.append()` and `<list> += <value>` in the same file?

Reply:
I used += to concat a list, and append for a single item

Reply:
We would normally use `extend` in the other examples, maybe go with that for consistency?

Reply:
personally if I was using `.append()`, I'd use `.extend()` but also not a big deal